src/main/java/net/earthcomputer/clientcommands/command/AreaStatsCommand.java

```diff
@@ -0,0 +1,205 @@
+import com.mojang.brigadier.exceptions.SimpleCommandExceptionType;
+import com.mojang.brigadier.tree.LiteralCommandNode;
+import net.earthcomputer.clientcommands.render.RenderQueue;
+import net.minecraft.block.Block;
+import net.minecraft.client.MinecraftClient;
+import net.minecraft.client.world.ClientWorld;
+import net.minecraft.server.command.ServerCommandSource;
+import net.minecraft.text.TranslatableText;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.Box;
+import net.minecraft.world.chunk.ChunkManager;
+import net.minecraft.world.chunk.WorldChunk;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Predicate;
+import java.util.stream.StreamSupport;
+
+import static net.earthcomputer.clientcommands.command.ClientCommandManager.*;
+import static net.earthcomputer.clientcommands.command.arguments.ListArgumentType.*;
+import static net.earthcomputer.clientcommands.command.arguments.SimpleBlockPredicateArgumentType.*;
+import static net.minecraft.command.argument.BlockPosArgumentType.*;
+import static net.minecraft.server.command.CommandManager.*;
+
+public class AreaStatsCommand {
+
+    private static final SimpleCommandExceptionType NOT_LOADED_EXCEPTION = new SimpleCommandExceptionType(new TranslatableText("commands.careastats.notLoaded"));
+
+    public static void register(CommandDispatcher<ServerCommandSource> dispatcher) {
+        addClientSideCommand("careastats");
+
+        LiteralCommandNode<ServerCommandSource> careastats = dispatcher.register(literal("careastats"));
+        dispatcher.register(literal("careastats")
+                .then(argument("pos1", blockPos())
+                        .then(argument("pos2", blockPos())
+                                .then(argument("predicates", list(blockPredicate(), 1))
+                                        .executes(ctx -> areaStats(ctx.getSource(), getBlockPos(ctx, "pos1"), getBlockPos(ctx, "pos2"), getList(ctx, "predicates"))))
+                                .executes(ctx -> areaStats(ctx.getSource(), getBlockPos(ctx, "pos1"), getBlockPos(ctx, "pos2"), block -> !block.getDefaultState().isAir())))));
+    }
+
+    private static int areaStats(ServerCommandSource source, BlockPos pos1, BlockPos pos2, Predicate<Block> blockPredicate) throws CommandSyntaxException {
+        return areaStats(source, pos1, pos2, Collections.singletonList(blockPredicate));
+    }
+
+    private static int areaStats(ServerCommandSource source, BlockPos pos1, BlockPos pos2, List<Predicate<Block>> blockPredicates) throws CommandSyntaxException {
+        final ClientWorld world = MinecraftClient.getInstance().world;
+        final ChunkManager chunkManager = world.getChunkManager();
+        if (!chunkManager.isChunkLoaded(pos1.getX() >> 4, pos1.getZ() >> 4) || !chunkManager.isChunkLoaded(pos2.getX() >> 4, pos2.getZ() >> 4)) {
+            throw NOT_LOADED_EXCEPTION.create();
+        }
```

Comment:
This method assumes that the loaded area is a square

Reply:
Good point. I'll check each chunk individually I guess.